docs/facebook-example.rst

```diff
@@ -97,36 +146,78 @@ To make this happen, since we've written the tasks already, we just need to form
-    :caption: workflow.py/workflow
-    :lines: 46-58
-    :lineno-start: 46
-    :linenos:
+.. testcode::
+
+   workflow = {
+       'mode': 'workflow',
+       'inputs': [
+           {'name': 'G',
+            'type': 'graph',
+            'format': 'adjacencylist'}
+       ],
+       'outputs': [
+           {'name': 'result_graph',
+            'type': 'graph',
+            'format': 'networkx'}
+       ]
+   }
 
 Now we need to add our tasks to the workflow, which is pretty straightforward since we've defined them in the previous steps.
 
-.. literalinclude:: static/facebook-example.py
-    :caption: workflow.py/workflow_tasks
-    :lines: 60-63
-    :lineno-start: 60
-    :linenos:
+.. testcode::
 
-Finally, we need to add the red arrows within the workflow, telling Romanesco how the inputs and outputs are going to flow from each task. These are called *connections* in Romanesco.
+   workflow['steps'] = [{'name': 'most_popular',
+                         'task': most_popular_task},
+                        {'name': 'find_neighborhood',
+                         'task': find_neighborhood_task}]
 
-.. literalinclude:: static/facebook-example.py
-    :caption: workflow.py/workflow_connections
-    :lines: 65-80
-    :lineno-start: 65
-    :linenos:
+Finally, we need to add the red arrows within the workflow, telling Romanesco how the inputs and outputs are going to flow from each task. These are called *connections* in Romanesco.
 
+.. testcode::
+
+   workflow['connections'] = [
+       {'name': 'G',
+        'input_step': 'most_popular',
+        'input': 'G'},
+       {'output_step': 'most_popular',
+        'output': 'G',
+        'input_step': 'find_neighborhood',
+        'input': 'G'},
+       {'output_step': 'most_popular',
+        'output': 'most_popular_person',
+        'input_step': 'find_neighborhood',
+        'input': 'most_popular_person'},
+       {'name': 'result_graph',
+        'output': 'subgraph',
+        'output_step': 'find_neighborhood'}
+   ]
 
 We now have a complete workflow! Let's run this, and write the final data to a file.
 
-.. literalinclude:: static/facebook-example.py
-    :caption: workflow.py/run
-    :lines: 82-91
-    :lineno-start: 82
-    :linenos:
+.. testcode::
+
+   with open('docs/static/facebook-sample-data.txt') as infile:
+       output = romanesco.run(workflow,
+                              inputs={'G': {'format': 'adjacencylist',
+                                            'data': infile.read()}},
+                              outputs={'result_graph': {'format': 'networkx.json'}})
+
+   with open('data.json', 'wb') as outfile:
+       outfile.write(output['result_graph']['data'])
+
+.. testoutput::
+   :hide:
+
+   --- beginning: most_popular ---
+   --- finished: most_popular ---
+   --- beginning: find_neighborhood ---
+   --- finished: find_neighborhood ---
+
+.. doctest::
+   :hide:
+
+   >>> open('data.json').read() == open('docs/static/data.json').read()
+   True
```

Comment:
This test is a little fragile in that it relies on the order of the keys in the serialized object, which is technically undefined.